src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -75,51 +83,140 @@ public async Task<(Error, IEnumerable<GitHubUser>)> GetUsersByCommit(string repo
+                                                email = "",
+                                                user = new
+                                                {
+                                                    databaseId = 0,
+                                                    login = "",
+                                                    name = "",
+                                                },
+                                            },
+                                        },
+                                    },
+                                },
+                            },
+                        },
+                    },
+                });
+
+                if (grahpApiResponse.errors != null && grahpApiResponse.errors.Any())
+                {
+                    return (Errors.GitHubApiFailed(_url, grahpApiResponse.errors.First().message), null);
+                }
+
+                var githubUsers = new List<GitHubUser>();
+                if (grahpApiResponse.data?.repository?.@object?.history?.nodes != null)
+                {
+                    foreach (var node in grahpApiResponse.data?.repository?.@object?.history?.nodes)
+                    {
+                        if (node.author != null)
+                        {
+                            githubUsers.Add(new GitHubUser
+                            {
+                                Id = node.author.user?.databaseId,
+                                Login = node.author.user?.login,
+                                Name = node.author.user?.name,
+                                Emails = new[] { node.author.email },
+                            });
+                        }
+                    }
+                }
+
+                return (null, githubUsers);
             }
             catch (RateLimitExceededException ex)
             {
-                _rateLimitError = Errors.GitHubApiFailed(apiDetail, ex);
+                _rateLimitError = Errors.GitHubApiFailed(_url, ex.Message);
                 return (_rateLimitError, null);
             }
             catch (Exception ex)
             {
-                LogAbuseExceptionDetail(apiDetail, ex);
-                return (Errors.GitHubApiFailed(apiDetail, ex), null);
+                LogAbuseExceptionDetail(_url, ex);
+                return (Errors.GitHubApiFailed(_url, ex.Message), null);
             }
-        }
 
-        private static GitHubUser ToGitHubUser(GitHubCommit commit)
-        {
-            return new GitHubUser
+            HttpRequestMessage CreateHttpRequest(HttpContent content)
             {
-                Id = commit.Author?.Id,
-                Login = commit.Author?.Login,
-                Name = commit.Commit.Author.Name,
-                Emails = new[] { commit.Commit.Author.Email },
-            };
+                var message = new HttpRequestMessage();
+                message.Headers.Add("User-Agent", "DocFX");
```

Comment:
prefer `HttpClient.DefaultRequestHeaders` for user agent and authorization, with that, you can simply call `HttpClient.PostAsync` without this helper function.

Reply:
Only one method is using this http client, and i believe not all are using bearer token, so I would like keep current way.